docs/hooks.md

```diff
@@ -0,0 +1,144 @@
+# Hooks
+
+Hooks are registered with the `fastify.graphql.addHook` method and allow you to listen to specific events in the GraphQL request/response lifecycle. You have to register a hook before the event is triggered, otherwise the event is lost.
+
+By using hooks you can interact directly with the GraphQL lifecycle of Mercurius. There are GraphQL Request hooks:
+
+- [GraphQL Request Hooks](#graphql-request-hooks)
+  - [preParsing](#preparsing)
+  - [preValidation](#prevalidation)
+  - [preExecution](#preexecution)
+  - [preGatewayExecution](#pregatewayexecution)
+  - [onResolution](#onresolution)
+  - [Manage Errors from a hook](#manage-errors-from-a-hook)
+  - [Add errors to the GraphQL response from a hook](#add-errors-to-the-graphql-response-from-a-hook)
+
+**Notice:** these hooks are only supported with `async`/`await` or returning a `Promise`.
+
+## GraphQL Request Hooks
+
+It is pretty easy to understand where each hook is executed by looking at the [lifecycle page](lifecycle.md).<br>
+
+There are four different hooks that you can use in GraphQL Request *(in order of execution)*:
+
+When registering hooks, you must wait for Mercurius to be registered in Fastify.
+
+```js
+await fastify.ready()
+```
+
+### preParsing
+
+If you are using the `preParsing` hook, you can access the GraphQL query string before it is parsed. It receives the schema and context objects as other hooks.
+
+For instance, you can register some tracing events:
+
+```js
+fastify.graphql.addHook('preParsing', async (schema, source, context) => {
+  await registerTraceEvent()
+})
+```
+
+### preValidation
+
+By the time the `preValidation` hook triggers, the query string has been parsed into a GraphQL Document AST.
+
+```js
+fastify.addHook('preValidation', async (schema, document, context) => {
```

Comment:
should this be `fastify.graphql.addHook`? only the first example in this document has that, not the others

Reply:
You're correct, yep - I'll submit a PR to fix this over lunch!